src/openforms/js/components/admin/form_design/RegistrationFields.js

```diff
@@ -51,10 +43,16 @@ const RegistrationFields = ({
                     name="form.registrationBackendOptions"
                     label={<FormattedMessage defaultMessage="Registration backend options" description="Registration backend options label" />}
                 >
-                    <TextInput
-                        value={JSON.stringify(backendOptions || {})}
-                        onChange={onBackendOptionsChange}
-                        maxLength="1000" />
+                    {backendOptionsForms[selectedBackend] ?
+                        <Form
+                            schema={backendOptionsForms[selectedBackend]}
+                            formData={backendOptions}
+                            onChange={({ formData }) => onChange({target: {name: 'form.registrationBackendOptions', value: formData}})}
+                            children={true}
+                        />
+                        :
+                        <></>
```

Comment:
```suggestion
                        null
```
Use null to clearly indicate nothing is to be rendered, keeps it consistent with other components.